diff -up mx-1.4.7/mx/mx-deform-texture.c.nogl mx-1.4.7/mx/mx-deform-texture.c
--- mx-1.4.7/mx/mx-deform-texture.c.nogl        2012-08-28 18:01:17.876523110 -0400
+++ mx-1.4.7/mx/mx-deform-texture.c     2012-08-28 19:07:34.062166545 -0400
@@ -493,9 +493,9 @@ mx_deform_texture_class_init (MxDeformTe
 static void
 mx_deform_texture_init_arrays (MxDeformTexture *self)
 {
-  GLushort *idx, *bf_idx;
+  gushort *idx, *bf_idx;
   gint x, y, direction;
-  GLushort *static_indices, *static_bf_indices;
+  gushort *static_indices, *static_bf_indices;
   MxDeformTexturePrivate *priv = self->priv;

   mx_deform_texture_free_arrays (self);
@@ -503,8 +503,8 @@ mx_deform_texture_init_arrays (MxDeformT
   priv->n_indices = (2 + 2 * priv->tiles_x) *
                     priv->tiles_y +
                     (priv->tiles_y - 1);
-  static_indices = g_new (GLushort, priv->n_indices);
-  static_bf_indices = g_new (GLushort, priv->n_indices);
+  static_indices = g_new (gushort, priv->n_indices);
+  static_bf_indices = g_new (gushort, priv->n_indices);

 #define MESH_INDEX(X, Y) (Y) * (priv->tiles_x + 1) + (X)

diff -up mx-1.4.7/mx/mx-texture-frame.c.nogl mx-1.4.7/mx/mx-texture-frame.c
--- mx-1.4.7/mx/mx-texture-frame.c.nogl 2012-08-28 19:18:45.691443210 -0400
+++ mx-1.4.7/mx/mx-texture-frame.c      2012-08-28 19:18:57.009446896 -0400
@@ -200,7 +200,7 @@ mx_texture_frame_paint (ClutterActor *se


   {
-    GLfloat rectangles[] =
+    gfloat rectangles[] =
     {
       /* top left corner */
       0, 0,
